src/hostManager/hostManager.ts

```diff
@@ -68,14 +68,27 @@ export class ClientHostManager {
     return { host: new URL(host), prefix };
   }
 
-  public getItemAsURL(context: Context, itemId: string) {
+  public getItemAsURL(
+    context: Context,
+    itemId: string,
+    qs: { [key: string]: string | number | boolean } = {},
+  ) {
     const host = this.getHost(context);
     const prefix = this.getPrefix(context);
     const url = new URL(`${prefix}/${itemId}`, host.origin);
+
+    for (const [k, v] of Object.entries(qs)) {
+      url.searchParams.set(k, v.toString());
```

Comment:
Just a note about `set`, it will override existing keys if duplicated keys are given. To add the same key multiple times use `append`. I don't think it is an issue here, but I just thought I would mention it.